### PR TITLE
drop deprecated TrainResult

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

removing deprecated class `TrainResult`, this shall be already removed in 1.0

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes [if needed]?
- [x] Did you write any new necessary tests [no need for typos, docs]? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:  

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone are aligned!**
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by gianscarpe — APPROVED** [review]
lgtm, I did the same in `test_cpu` in my PR #4820

**Comment by Borda** [comment]
> lgtm, I did the same in `test_cpu` in my PR #4820

nice, but yours is doing even more things, this is just removal, so no worries, we can lads both 🐰 
btw, I see that yours is a destination for the master...

**Review by ydcjeff — APPROVED** [review]

**Review by SkafteNicki — APPROVED** [review]

File: tests/models/test_cpu.py [hunk 1]
```diff
@@ -379,151 +378,3 @@ def train_dataloader(self):
     result = trainer.fit(model)
 
     assert result == 1, 'training failed to complete'
-
-
-def test_tbptt_cpu_model_result(tmpdir):
```
Comment (carmocca) [inline tests/models/test_cpu.py:384]: Why are these gone?

File: tests/base/model_train_steps.py [hunk 2]
```diff
@@ -62,90 +60,23 @@ def training_step__inf_loss(self, batch, batch_idx, optimizer_idx=None):
                 output /= 0
         return output
 
-    def training_step_result_obj_dp(self, batch, batch_idx, optimizer_idx=None):
+    def training_step__result_obj_dp(self, batch, batch_idx, optimizer_idx=None):
+        self.training_step_called = True
```
Comment (carmocca) [inline tests/base/model_train_steps.py:64]: Is adding this necessary?

Comment (Borda) [reply tests/base/model_train_steps.py:64]: it is just renamed to follow the guideline for tests...

File: tests/models/test_cpu.py (hunk 1 above)
Comment (Borda) [reply tests/models/test_cpu.py:384]: do not ask me :D you try git blame who removed/rewrote relevant tests :]

File: tests/base/model_train_steps.py (hunk 2 above)
Comment (carmocca) [reply tests/base/model_train_steps.py:64]: I mean adding `self.training_step_called = True`

**Review by awaelchli — APPROVED** [review]

**Review by s-rog — APPROVED** [review]

**Comment by Borda** [comment]
needed for #5005
